controllers/userMigrations.js

```diff
@@ -0,0 +1,42 @@
+const userQuery = require("../models/userMigrations");
+const logger = require("../utils/logger");
+const { getRandomIndex } = require("../utils/helpers");
+const userColors = 10;
+/**
+ * Returns the lists of usernames where default colors were added
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const addDefaultColors = async (req, res) => {
+  try {
+    const { usersArr, userModel } = await userQuery.addDefaultColors();
+    const migratedUsers = [];
+    const updateUserPromises = [];
+
+    for (const user of usersArr) {
+      const colors = user.colors ?? {};
+      if (user.colors === undefined) {
+        const userColorIndex = getRandomIndex(userColors);
+        colors.color_id = userColorIndex;
+        updateUserPromises.push(userModel.doc(user.id).set({ ...user, colors }));
+        migratedUsers.push(user.username);
+      }
+    }
+
+    await Promise.all(updateUserPromises);
+
```

Comment:
What do you think on considering using firestore batched writes or transactions

https://firebase.google.com/docs/firestore/manage-data/transactions

Reply:
yes makes sense

Reply:
updated with batch